internal/clients/interface.go

```diff
@@ -111,10 +111,11 @@ type Azure interface {
 	EnsureResourceGroup(ctx context.Context, name string, location string) (*string, error)
 
 	// CreateVM creates Azure virtual machine.
-	// Most of the parameters are constant for now.
-	// location - to deploy into
-	// imageName - the imageID will be inferred as /subscriptions/{subscriptionID}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/images/{imageName}
-	CreateVM(ctx context.Context, location string, resourceGroupName string, imageName string, pubkey *models.Pubkey, instanceType InstanceTypeName, vmName string, userData []byte) (*string, error)
+	CreateVM(ctx context.Context, params AzureInstanceParams, vmName string) (*string, error)
+
+	// CreateVMs creates multiple Azure virtual machines
+	// Returns array of instance IDs and error if something went wrong
+	CreateVMs(ctx context.Context, instanceParams AzureInstanceParams, amount int64, vmNamePrefix string) (vmIds []*string, err error)
 }
```

Comment:
Thanks for not exposing wait functions. I will say that CreateVM = CreateVMs with amount=1 so we might not want to expose the CreateVM at all. Remember, more functions = more tests = more states to cover.

Reply:
in a new commit, I've removed the `CreateVM` function all together, I believe it is not very usefull anymore.